app/auth/middleware.js

```diff
@@ -38,6 +30,18 @@ function processAuthResponse(defaultLocations = []) {
   }
 }
 
+function _setSessionData(session, token, postAuthRedirect) {
+  const tokenData = decodeAccessToken(token)
+
+  session.authExpiry = tokenData.exp
+  session.postAuthRedirect = postAuthRedirect
+  session.user = new User({
+    name: tokenData.user_name,
+    roles: tokenData.authorities,
+    locations: userLocationService.getUserLocations(token),
```

Comment:
Does this not require `await` as the call is async?

Probably neater to leave the locations call on line 19, just inside the `try/catch`, like before. And can then leave the line above the same as it was.

Reply:
Ah, the test is passing because you're not returning a promise, which differs from what `getUserLocations` actually returns.

If you look how the test worked for the reference data stub you should be able to copy that to test promise rejection and success.

Reply:
I've fixed the test and added test cases for failure.